testsuite/gateways/apicast/selfmanaged.py

```diff
@@ -18,8 +15,16 @@ class NoSuitableApicastError(Exception):
     """Raised if no deployment method is found"""
 
 
+# pylint: disable=too-many-instance-attributes
 class SelfManagedApicast(AbstractApicast):
```

Comment:
I would change this to inherit from `OpenShiftApicast` to not break typing. As the real class, it will return will be a subclass of `OpenShiftApicast`

Reply:
My thought is that this can be also `DockerApicast` one day. Thoughts? (not necessarily tomorrow or next week but generally speaking)

Reply:
I was planning to rely on that assumptions while making TLS for OperatorApicast, so at least for now it would be better to have it bound to `OpenshiftApicast`, however we can reevaluate when we decide we want to implement `DockerApicast`.